### DISCUSSION TITLE
scraping posts, their comments(not just the number of comments) and Likes, shares

### DISCUSSION THREAD

**Discussion by salma-eisa** [kevinzg/facebook-scraper#836] [Q&A]
I'm doing work for a Ph.D. thesis. where I have to collect data and then use it for comment classification. I'm trying to use this program but I don't seem to know how to get the comments.

**Comment by neon-ninja** [comment]
Set `options={"comments": True}`. Like: `get_posts(page_name, options={"comments": True})`

**Comment by salma-eisa** [comment]
Hello and thank you for your help. I tried the following :
from facebook_scraper import get_posts
get_posts('ameedcoffee', options={"comments": True})
but the only thing i got was :

generator object FacebookScraper._generic_get_posts at 0x000002D7F6EF0B30

am i doing something wrong?


Thanks

**Reply by neon-ninja** [reply]
`get_posts` returns a generator, that you can iterate over (typically with a for loop). Try:

```python
from pprint import pprint
for post in get_posts('ameedcoffee', options={"comments": True}):
  pprint(post)
```
